src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,67 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+
 public class WorkWithFile {
+    public static final int INDEX_OF_CATEGORY = 0;
+    public static final int INDEX_OF_VALUE = 1;
+    public static final int LENGTH_OF_ARRAY = 2;
+    public static final int SUPPLY = 0;
+    public static final int BUY = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            int[] statistics = parseInformationFromFile(fromFileName);
+            String report = buildReport(statistics);
+            writeReport(toFileName, report);
+        } catch (IOException e) {
+            throw new RuntimeException("I/O error while processing files: "
+                    + fromFileName + " or " + toFileName, e);
+        }
+    }
+
+    private int[] parseInformationFromFile(String fromFileName) throws IOException {
+        int[] result = new int[LENGTH_OF_ARRAY];
+        BufferedReader reader = new BufferedReader(new FileReader(fromFileName));
+        String line;
+
+        while ((line = reader.readLine()) != null) {
+            String[] separatedLine = line.trim().split(",");
+            try {
+                switch (separatedLine[INDEX_OF_CATEGORY]) {
+                    case "supply" -> result[SUPPLY]
+                            += Integer.parseInt(separatedLine[INDEX_OF_VALUE]);
```

You rely on Integer.parseInt without handling NumberFormatException. The checklist says not to ignore exceptions ("Don't ignore exceptions"). Wrap parse in try/catch and rethrow a RuntimeException with a clear message including the offending line and the original cause so failures are diagnosable. (Checklist: "Don't ignore exceptions") .